src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/dataobject.cs

```diff
@@ -2467,41 +2462,29 @@ public void SetData(Object data)
 
             public string[] GetFormats(bool autoConvert)
             {
+                IEnumFORMATETC enumFORMATETC = EnumFormatEtcInner(DATADIR.DATADIR_GET);
+                List<string> formats = [];
```

Comment:
How about re-define formats with `HashSet<string>`, which can remove the `GetDistinctStrings` method and be faster?

Reply:
@h3xds1nz already discussed this in the description, the number of items is too low to justify the overhead of hashset.

Reply:
@lindexi  I further checked if using a HastSet out of the box and thus saving ourselves the uniqueness check (+method) could prove itself useful but I couldn't come up with a use-case where the overhead is justified even this way. Thanks for the inquiry though, love it.

Reply:
Thank you for your contribution.